### PR TITLE
Add AttributionControl

### PR DESCRIPTION
**Pull request by Pessimistress**
#1129

### CODE REVIEW

File: src/components/attribution-control.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+const propTypes = Object.assign({}, mapControlPropTypes, {
+  toggleLabel: PropTypes.string,
+  className: PropTypes.string,
+  style: PropTypes.object,
+  compact: PropTypes.bool,
+  customAttribution: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)])
+});
+
+const defaultProps = Object.assign({}, mapControlDefaultProps, {
```
Comment (macrigiuseppe) [inline src/components/attribution-control.js:15]: how do you feel about destructuring here:
```
const defaultProps ={...mapControlDefaultProps, {
  className: '',
  toggleLabel: 'Toggle Attribution'
}};
```

File: src/components/attribution-control.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  useEffect(() => {
+    if (!compact && showCompact) {
+      setShowCompact(false);
+    }
+  }, [compact]);
+
+  const toggleAttribution = useCallback(() => setShowCompact(value => !value), []);
+
+  return (
+    <div style={{position: 'absolute', ...props.style}} className={props.className}>
```
Comment (macrigiuseppe) [inline src/components/attribution-control.js:72]: take advantage of useMemo the same way you did in other PRs
```
useMemo(() => ({position: 'absolute', ...(props.style || {})}), [props.style])
```
I would make sure prop.style is not null

File: src/components/attribution-control.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      <div
+        ref={containerRef}
+        aria-pressed={showCompact}
+        className={`mapboxgl-ctrl mapboxgl-ctrl-attrib ${compact ? 'mapboxgl-compact' : ''} ${
```
Comment (macrigiuseppe) [inline src/components/attribution-control.js:76]: if we are planning on having several instances of class name manipulation we should think about using something like `classnames` to better handle these cases

**Review by macrigiuseppe — APPROVED** [review]

File: src/components/attribution-control.js (hunk 2 above)
Comment (Pessimistress) [reply src/components/attribution-control.js:72]: I believe the spread operator supports null/undefined (similar to `Object.assign`)

Comment (Pessimistress) [reply src/components/attribution-control.js:72]: Will address in #1297

**Pull request merged**